patches/chromium/printing.patch

```diff
@@ -117,28 +117,34 @@ index 81cb954f02363c829947dde830f340a761c80a77..2a26bb6d5367337fa13da07cb0758868
  }
  
  PrintViewManagerBase::~PrintViewManagerBase() {
-@@ -123,7 +126,8 @@ PrintViewManagerBase::~PrintViewManagerBase() {
+@@ -123,7 +126,9 @@ PrintViewManagerBase::~PrintViewManagerBase() {
    DisconnectFromCurrentPrintJob();
  }
  
 -bool PrintViewManagerBase::PrintNow(content::RenderFrameHost* rfh) {
 +bool PrintViewManagerBase::PrintNow(content::RenderFrameHost* rfh,
-+                                    CompletionCallback callback) {
++                                    CompletionCallback callback,
++                                    base::Value settings) {
    DisconnectFromCurrentPrintJob();
  
    // Don't print / print preview interstitials or crashed tabs.
-@@ -131,7 +135,9 @@ bool PrintViewManagerBase::PrintNow(content::RenderFrameHost* rfh) {
+@@ -131,7 +136,14 @@ bool PrintViewManagerBase::PrintNow(content::RenderFrameHost* rfh) {
      return false;
  
    SetPrintingRFH(rfh);
 -  GetPrintRenderFrame(rfh)->PrintRequestedPages();
 +  callback_ = std::move(callback);
-+  // FIXME: pass 'silent' through from the caller..?
-+  GetPrintRenderFrame(rfh)->PrintRequestedPages(false /* silent */);
++
++  if (!callback_.is_null()) {
++    registrar_.Add(this, chrome::NOTIFICATION_PRINT_JOB_EVENT,
++            content::NotificationService::AllSources());
++  }
++
++  GetPrintRenderFrame(rfh)->PrintRequestedPages(false /* silent */, std::move(settings));
```

Comment:
do you need to pass `silent` here..?

Reply:
whoops, i forgot to plumb this through correctly 🙈 
We do though - Chromium doesn't have a concept of silent printing so it's not included in the bespoke print settings the way that the other options are

Reply:
this is tested to be good now ✅